daemon/daemon.go

```diff
@@ -1120,18 +1123,15 @@ func (daemon *Daemon) waitForStartupDone() {
 	<-daemon.startupDone
 }
 
-func (daemon *Daemon) shutdownContainer(c *container.Container) error {
+func (daemon *Daemon) shutdownContainer(ctx context.Context, c *container.Container) error {
 	stopTimeout := c.StopTimeout()
 
 	// If container failed to exit in stopTimeout seconds of SIGTERM, then using the force
-	if err := daemon.containerStop(c, stopTimeout); err != nil {
-		return fmt.Errorf("Failed to stop container %s with error: %v", c.ID, err)
+	if err := daemon.containerStop(ctx, c, stopTimeout); err != nil {
+		return fmt.Errorf("%s: failed to stop container: %w", c.ID, err)
```

Comment:
* Yes, we need to include the container ID in this error. Proof: `errNoSuchProcess` (returned from `killProcessDirectly()` via `daemon.Kill()`) is decorated with the PID, not the container ID.
* None of the branches of the `daemon.containerStop()` call graph returns a `github.com/pkg/errors` error value so there is no original stack trace to preserve using `errors.Wrap()`.
* This error does not get returned to API clients so it will not be seen at the command line. `daemon.shutdownContainer()` is called in exactly two places, both of which handle the returned error by logging it to logrus.